internal/aws/awsutil/conn.go

```diff
@@ -103,6 +105,21 @@ func GetAWSConfig(ctx context.Context, logger *zap.Logger, settings *AWSSessionS
 		return aws.Config{}, err
 	}
 
+	if settings.RoleARN != "" {
```

Comment:
Is it possible to add a test for this? e.g. by mocking both the STS and the CloudWatch API

It was broken once before, adding a test will help avoid breaking it once again

Reply:
They suggest using mocking https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/unit-testing.html 

but I am not sure this can validate the auth mechanism.